woocommerce/woocommerce
# Fix: variations exported as draft were imported as draft

### All Submissions:






### Changes proposed in this Pull Request:



When a product export that involves a variable product in draft status is performed, the "published" row of the generated CSV has a status of "draft" not only for the main variable product but also for the variations, even though the variations themselves are always in "published" state (otherwise they aren't shown when editing the product).

The product import code is supposed to handle variations as a special case, creating them a status of "published" even if they were set as "draft" in the CSV; but due to a bug (a piece of code that assumed that `1 === 1.0` was true) that wasn't happening. This pull request fixes that.

Closes #34962.



### How to test the changes in this Pull Request:




1. Go to WooCommerce - Products and create a variable product with some variations. Don't publish it; instead, click on "Save as draft".
2. Generate a product export (WooCommerce - Products - "Export" button) that includes all the columns. Optionally, first create a new product category and assing it to the product you just created, then export only that category, so that the generated CSV file contains only that product.
3. Delete the product.
4. Import the CSV file you just created ((WooCommerce - Products - "Import" button).
5. Open the product details and verify that you can see the variations. Without this fix the main product would have been imported but variations would remain invisible (they would have a status of "draft" in the database).

### Other information:


### FOR PR REVIEWER ONLY:
